rir/src/compiler/analysis/loop_detection.cpp

```diff
@@ -0,0 +1,106 @@
+#include "loop_detection.h"
+#include "../util/visitor.h"
+
+namespace rir {
+namespace pir {
+
+bool LoopDetection::Loop::comesBefore(Instruction* a, Instruction* b) const {
+    if (a->bb() == b->bb()) {
+        return a->id().idx() < b->id().idx();
```

Comment:
just a drive-by comment.

this is very (!) slow... since this scans the instruction array for both instructions. I think we should probably remove idx() from the api. (I think we only use it for printing, where I guess we don't care that it is slow).

Can you pass iterators instead of instructions? because then it is just `a < b`.

Reply:
Thanks for the catch -- this would explain why it's so slow.

For now, I might just rip it out of this PR, and only add it in the loop peeling PR. The thing is, I find this `comesBefore` and `ordering` vector to be really hacky, and I'd like to find a better solution.